dagger-core/src/main/java/com/gotocompany/dagger/core/sink/SinkOrchestrator.java

```diff
@@ -106,6 +106,9 @@ protected Properties getProducerProperties(Configuration configuration) {
             kafkaProducerConfigs.setProperty(Constants.SINK_KAFKA_COMPRESSION_TYPE_KEY, Constants.SINK_KAFKA_COMPRESSION_TYPE_DEFAULT);
             kafkaProducerConfigs.setProperty(Constants.SINK_KAFKA_MAX_REQUEST_SIZE_KEY, Constants.SINK_KAFKA_MAX_REQUEST_SIZE_DEFAULT);
         }
+        String lingerMs = configuration.getString(Constants.SINK_KAFKA_LINGER_MS_KEY, Constants.SINK_KAFKA_LINGER_MS_DEFAULT);
```

Comment:
can you add a check here that if the value of this config is not a valid long , then throw an IllegalArgumentException, so that it fails during job creation itself, or you can just set it to 0 when it is not a valid long

Reply:
This is also performed internally by the Kafka Producer as well. Ref: [Linger ms inside producer client](https://github.com/apache/kafka/blob/trunk/clients/src/main/java/org/apache/kafka/clients/producer/KafkaProducer.java#L550) which internally uses (Config)[https://github.com/a0x8o/kafka/blob/master/clients/src/main/java/org/apache/kafka/common/protocol/types/Struct.java#L288]. Since, the contract for the property from the kafka layer is string. Unsure if we should be checking it explicitly. WDYT ?

Reply:
yeah it is checked by kafka but that is after job submission, so if any invalid values are passed the job will keep restarting, but it will be better if it fails during job creation itself

Reply:
Ohh, understood. Have added a validate method, and a corresponding test to check it's correctness.